polkadot/runtime/westend/src/lib.rs

```diff
@@ -1523,9 +1558,27 @@ pub type Migrations = migrations::Unreleased;
 pub mod migrations {
 	use super::*;
 
+	parameter_types! {
+		pub const ImOnlinePalletName: &'static str = "ImOnline";
+	}
+
+	/// Upgrade Session keys to exclude `ImOnline` key.
+	/// When this is removed, should also remove `OldSessionKeys`.
+	pub struct UpgradeSessionKeys;
+	impl frame_support::traits::OnRuntimeUpgrade for UpgradeSessionKeys {
+		fn on_runtime_upgrade() -> Weight {
```

Comment:
Please add proper `pre` and `post` runtime upgrade checks.

Reply:
Can you please elaborate on what exactly you want me to check? Usually, we check the storage version change in `pre` and `post`, but the `SessionKeys` are not versioned. The session keys themselves are not available at the moment IIUC so it also doesn't make much sense to encode/decode them in `pre`/`post`. Checking the runtime spec number is redundant as it's not a predicate for this migration anyway, and it's checked inside the migration. So I'm a bit lost with what to check :man_shrugging:

Reply:
Anyway, I've tried to make it as best I could. Please check if it's ok.